waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -205,6 +205,7 @@
                                               ;; we don't include the exit code in failed instances that were killed by k8s.
                                               (not (killed-by-k8s? newest-failure))
                                               (assoc :exit-code (:exitCode newest-failure)))]
+                (scheduler/log-service-instance newest-failure-instance :fail)
                 (swap! service-id->failed-instances-transient-store
                        update-in [service-id] assoc newest-failure-id newest-failure-instance)))))))
```

Comment:
We should wrap the `service-id->failed-instances-transient-store` update in a function and that function can log the service instance. E.g. that function can be reused in the marathon scheduler below.

Reply:
Done